lib/cocoapods/generate/installer.rb

```diff
@@ -92,19 +97,24 @@ def open_app_project(recreate: false)
       # @return [Xcodeproj::Project]
       #
       def create_app_project
-        app_project = open_app_project(recreate: true)
+        app_project = open_app_project(recreate: !configuration.incremental_installation?)
+        # Do not re-integrate the "user" project if it already exists and we are performing an incremental installation.
+        return if app_project.path.exist? && configuration.incremental_installation?
```

Comment:
Couldn't there still be an incremental change that requires updating the user project?

Reply:
talked offline. CocoaPods will handle that internally for the user projects as well. However, I can remove this check and instead avoid writing the native targets if they already exist.

Reply:
updated and now we always write the user project and avoid re-creating its native targets.